kubernetes-model/kubernetes-model/src/main/java/io/fabric8/kubernetes/api/model/Quantity.java

```diff
@@ -46,86 +48,201 @@
 @Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage=true, builderPackage = "io.fabric8.kubernetes.api.builder", inline = @Inline(type = Doneable.class, prefix = "Doneable", value = "done"))
 public class Quantity  implements Serializable {
 
-    private String amount;
-    private String format;
-    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
+  private String amount;
+  private String format;
+  private Map<String, Object> additionalProperties = new HashMap<String, Object>();
 
-    /**
-     * No args constructor for use in serialization
-     *
-     */
-    public Quantity() {
-    }
+  /**
+   * No args constructor for use in serialization
+   *
+   */
+  public Quantity() {
+  }
 
-    /**
-     * Single argument constructor for setting amount.
-     *
-     * @param amount amount of quantity specified.
-     */
-    public Quantity(String amount) {
-        this.amount = amount;
-    }
+  /**
+   * Single argument constructor for setting amount.
+   *
+   * @param amount amount of quantity specified.
+   */
+  public Quantity(String amount) {
```

Comment:
If this is only used to deserialize it would probably be better to create a static factory/parser method and use that plus the use of `AbstractMap.SimpleEntry` is not really appropriate here.
```
* static Quantity parse(String formattedValue) {
    //... do your parsing stuff
    return new Quantity(amountStr, formatStr);
}
```

If the "intelligent" constructor is really necessary, then ditch the AbstractMap.SimpleEntry and use the parsed instance to prefill the new instance (kind of ugly too):
```
final Quantity parsedQuantity = parse(amount);
this.amount = parsedQuantity.getAmount();
this.format = parsedQuantity.getFormat();
```